src/test/java/ru/nms/TestUtils.java

```diff
@@ -0,0 +1,102 @@
+package ru.nms;
+
+import org.apache.commons.math3.linear.RealVector;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Random;
+
+import static ru.nms.TestDataGenerator.generateNVectors;
+import static ru.nms.utils.BallTreeUtils.*;
+import static ru.nms.utils.Constants.BALL_TREE_HEADER;
+import static ru.nms.utils.Constants.TEMP_DIR;
+
+public class TestUtils {
+
+    public static final Random RANDOM = new Random(0);
+    private static final int CHUNK_SIZE = Integer.MAX_VALUE;
+    public static String SOURCE_FILE_DIR = "source_files/";
+
+    public static void writeVectorsToFile(List<RealVector> vectors, String fullPath) throws IOException {
+        if (vectors.isEmpty()) {
+            throw new IllegalArgumentException("The list of vectors is empty.");
+        }
+
+        int dimension = vectors.get(0).getDimension();
+        long vectorCount = vectors.size();
+
+        try (RandomAccessFile file = new RandomAccessFile(fullPath, "rw");
+             FileChannel fileChannel = file.getChannel()) {
+            fileChannel.write(ByteBuffer.wrap(BALL_TREE_HEADER.getBytes(StandardCharsets.UTF_8)));
+
+            file.writeLong(vectorCount);
+            file.writeInt(dimension);
+            setPositionToStartOfVectorData(fileChannel, dimension);
+
+            for (RealVector vector : vectors) {
+                putVectorToFile(file, vector);
+            }
+        }
+    }
+
+    public static void cleanTempDir() {
+        File directory = new File(TEMP_DIR);
+        if (directory.exists() && directory.isDirectory()) {
+            File[] files = directory.listFiles();
+
+            if (files != null) {
+                for (File file : files) {
+                    if (file.isFile()) {
+                        if (!file.delete()) {
+                            System.out.println("Failed to delete: " + file.getName());
+                        }
+                    }
```

Comment:
```suggestion
                    if (file.isFile() && !file.delete()) {
                        System.err.println("Failed to delete: " + file.getName());
                    }
```

(`err` просто для красоты, но вообще см. коммент про логгинг)